pulp_container/app/viewsets.py

```diff
@@ -547,13 +552,9 @@ class ContainerRepositoryVersionViewSet(RepositoryVersionViewSet):
     parent_viewset = ContainerRepositoryViewSet
 
 
-class ContainerPushRepositoryViewSet(ReadOnlyRepositoryViewSet):
+class ContainerPushRepositoryViewSet(TagOperationsMixin, RepositoryViewSet):
```

Comment:
I'm afraid we cannot use RepositoyViewSet since it adds the DELETE, only list/get/post should be allowed.

Reply:
Since I cannot access `AsyncUpdateMixin` from `pulpcore` in order to define custom `ModifiableRepositoryViewSet`, I defined all allowed methods in `http_method_names` manually. This eventually forbids the DELETE operation.

Reply:
So `AsyncUpdateMixin` should become part of the plugin api? But we don't want "update" either on push repos.

Reply:
It will suffice to leave ReadOnlyRepositoryViewSet. This will make sure you cannot perform post/put/patch/delete on the repository object but will still enable POST action for tag which you have defined in TagOperationsMixin.